### ISSUE TITLE
screen reader is not announcing the search result under Pivot mode

### ISSUE THREAD

**Issue by msftedad** [ag-grid/ag-grid#6210]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Repro Steps**
1.Open URL:https://plnkr.co/edit/uFa3bMppk8tOXQLL?open=main.ts&preview by using the enter key.
2.The screen will open with Tool Panel Simple
3.Keyboard focus automatically moves on ‘Main.ts’ field.
4.Now Press ‘Ctrl+M' key then press 1 tab key, keyboard focus moves on 'Preview' control.
5.Turn on the narrator and press "E' shortcut key twice to navigate search field.
6. Search valid/invalid information and Verify the screen reader announcing search result information or not.

**Current behavior**
After entering value in search field, screen reader is not announcing the count for the search results under "Columns"
**observation**: After entering "age" in search edit field, results are displayed but not announcing result information.

**Expected behavior**
After entering value in search field, screen reader should announce the count of the search results under "Columns"
**Solution**: After entering "Age" screen reader should announce like 1 result found 

**Please tell us about your environment:**

**Test Environment:**  

OS Version: 22H2 OS Build 22621.674

* **AG Grid version:** 27.3.0

* **Browser:**
Edge(Chromium) Version 109.0.1495.2 (Official build) dev (64-bit)

* **Language:** [ES5]

https://user-images.githubusercontent.com/93735775/220620196-c5580c17-5640-44fc-bc99-4439ebc29b39.mp4

**Comment by kiril-matev** [comment]
Hello @msftedad,

Thank you for bringing this up with us. 

The functionality you described wasn't part of the requirements when we built this feature. This is why we have added a feature request in our backlog to track this requirement and implement it in the future.

You can track it with this reference:
AG-6541 Hello,

Thank you for bringing this up with us.

The functionality you described wasn't part of the requirements when we built this feature. This is why we have added a feature request in our backlog to track this requirement and implement it in the future.

You can track it with this reference:
**AG-6541 [Accessibility] Allow announcing the number of matching list items when using the search box in the columns/filter tool panel and set filter**

You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

Now that this requirement is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this feature is added in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/

Thanks again for bringing this up with us - we appreciate your time!

You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this requirement is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this feature is added in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!

**Comment by msftedad** [comment]
@kiril-matev Thank you for the update. We will get in touch with our engineering team on this and let you know if the added feature is fixing this issue.

**Comment by msftedad** [comment]
Hi @StephenCooper , as verified on OS Version: Version 22H2 (OS Build 22621.2283).Browser:Version119.0.2109.1 (Official build) dev (64-bit). The issue is still repro as screen reader is not announcing the count for the search results After entering value in search field. Can you please reopen this issue. Thanks

**Comment by kiril-matev** [comment]
Hi @msftedad,

This will be implemented in our next major release - AG Grid v31. It's scheduled for release by the end of November 2023.

Please upgrade to AG Grid v31 or later when it's released to resolve this.

**Comment by kiril-matev** [comment]
Hello @hoovercj,

I'm glad to hear you've upgraded to AG Grid 31.
 
This enhancement to accessibility support is default behavior - it doesn't need to be enabled and cannot be disabled. As this should be default behavior and search results should always be announced, there's no documentation specifically for this feature.
 
There are localization keys for this feature here:
https://www.ag-grid.com/javascript-data-grid/localisation/#creating-a-locale
 
    // ARIA Labels for the Side Bar 
    ariaColumnPanelList: 'Column List', 
    ariaFilterPanelList: 'Filter List',
    
        // Side Bar 
    columns: 'Columns', 
    filters: 'Filters',
 
You can modify these strings to customize the announcement. You can see the effect of modified localization keys on this announcement in this sample:
https://www.ag-grid.com/javascript-data-grid/localisation/#installing-a-locale
 
Please go through the test cases below to see this feature in action:
 
TC1 - columns tool panel - announce number of matches
Steps to illustrate
0. Open NVDA
1. Open https://www.ag-grid.com/example/
2. Click inside the search box of the columns tool panel
3. Type 'country'
Actual & Expected:
1/ Columns list filtered to only show one column - COUNTRY.
2/ NVDA announcement - 'column list 2 columns'
 
TC2 - columns tool panel - announce no matches
Steps to illustrate
Open NVDA
Open https://www.ag-grid.com/example/
Click inside the search box of the columns tool panel
Type 'yy'
Actual & Expected:
1/ Columns list filtered to no items
2/ NVDA Announcement - column list 0 columns
 
TC3 - filters tool panel
Steps to illustrate
Open NVDA
Open https://www.ag-grid.com/example/
Click inside the search box of the filter tool panel
Type 'country'
Actual & Expected:
1/ Filters list filtered to only show one filter - COUNTRY.
2/ NVDA announcement - filter list 1 filters
 
TC4 - filters tool panel - no matches announcement
Steps to illustrate
Open NVDA
Open https://www.ag-grid.com/example/
Click inside the search box of the filter tool panel
Type 'yy'
Actual & Expected:
1/ Filters list filtered to show no matches
2/ NVDA Announcement - filter list 0 filters
